src/main/kotlin/net/irisshaders/lilybot/extensions/util/PublicUtilities.kt

```diff
@@ -57,5 +71,249 @@ class PublicUtilities : Extension() {
 				}
 			}
 		}
+
+		/**
+		 * Nickname request command
+		 * @author NoComment1105
+		 */
+		ephemeralSlashCommand {
+			name = "nickname"
+			description = "The parent command for all nickname commands"
+
+			ephemeralSubCommand(::NickRequestArgs) {
+				name = "request"
+				description = "Request a new nickname for the server!"
+
+				check { anyGuild() }
+
+				action {
+					val actionLogId = getConfigPublicResponse("modActionLog") ?: return@action
+					val actionLog = guild?.getChannel(actionLogId) as GuildMessageChannelBehavior
+
+					val requester = user.asUser()
+					val requesterAsMember = requester.asMember(guild!!.id)
+					var actionLogEmbed: Message? = null
+
+					respond { content = "Nickname request sent!" }
+
+					actionLogEmbed = actionLog.createEmbed {
+						color = DISCORD_YELLOW
+						title = "Nickname Request"
+						timestamp = Clock.System.now()
+
+						field {
+							name = "User:"
+							value = "${requester.mention}\n${requester.asUser().tag}\n${requester.id}"
+							inline = false
+						}
+
+						field {
+							name = "Current Nickname:"
+							value = "`${requesterAsMember.nickname}`"
+							inline = false
+						}
+
+						field {
+							name = "Requested Nickname:"
+							value = "`${arguments.newNick}`"
+							inline = false
+						}
+					}.edit {
+						components {
+							ephemeralButton(row = 0) {
+								label = "Accept"
+								style = ButtonStyle.Success
+
+								action {
+									requesterAsMember.edit { nickname = arguments.newNick }
+
+									userDMEmbed(
+										requester.asUser(),
+										"Nickname Change Accepted in ${guild!!.asGuild().name}",
+										"Nickname updated from `${requesterAsMember.nickname}` to " +
+												"`${arguments.newNick}`",
```

Comment:
wouldn't requesterAsMember.nickname be equal to arguments.newNick, since you already changed their nickname?

Reply:
No, requesterAsMember variable was already assigned earlier in the command. It will still get the old nickname